garden_ai/garden.py

```diff
@@ -79,3 +81,65 @@ def _authenticate(self):
             on_refresh=self.auth_key_store.on_refresh,
         )
         return authorizer
+
+    def create_garden(self, **kwargs):
+        """Initialize and return a new Garden object
+
+        If invoked with keyword arguments corresponding to field names (e.g.
+        `title` or `creator`) are passed directly to the `Garden` model, making
+        it possible for the user to provide necessary info upfront to be parsed
+        and validated at initialization, or to incrementally provide field
+        values via simple assignment to attributes of the returned object
+        directly.
+
+        Parameters
+        ----------
+        **kwargs :
+            Metadata for the new Garden object. Keyword arguments matching
+            required or recommended fields will be coerced to the appropriate
+            type, if necessary.
+
+        Examples
+        --------
+
+            gc = GardenClient()
+            pea_garden = gc.create_garden(
+                creators=["Mendel, Gregor"], title="Experiments on Plant Hybridization"
+            )
+            pea_garden.year = 1863
+        """
+        return Garden(**kwargs)
+
+    def register_garden(self, G: Garden):
+        """Finalize the setup of a Garden object with user-populated fields
```

Comment:
please add more context. What does "finalize" mean? The method name has the word "register" -- what is being registered here? from the code, it looks like the key component is the minting of the DOI. Please make that clear in the docstring

Reply:
this is the most placeholder-y part of the code I think. my reasoning here was (copied from my convo with Logan in slack): if a user is invoking some `GardenClient` method to finalize/register their `Garden` object, that's a safe point to assume the user is "done" (or thinks they're done) providing metadata and seemed like a reasonable point to run a full suite of validation functions

Reply:
I think that design-wise this method is probably trying to do too much, and to be fully transparent I was kinda hoping "register" would be evocative but vague enough to not think too hard about what it meant

Reply:
Made it clearer in docstring that this is mainly for tying up loose metadata-ends